cmd/vic-machine/common/utils.go

```diff
@@ -31,3 +40,24 @@ func LogErrorIfAny(clic *cli.Context, err error) error {
 	log.Errorf("%s %s failed: %s\n", clic.App.Name, clic.Command.Name, errors.ErrorStack(err))
 	return cli.NewExitError("", 1)
 }
+
+// CheckUnsupportedChars returns an error if string contains special characters
+func CheckUnsupportedChars(s string) error {
+	re := regexp.MustCompile(unsuppCharsRegex)
+	return checkUnsupportedChars(s, re)
+}
+
+// CheckUnsupportedCharsDatastore returns an error if a datastore string contains special characters
+func CheckUnsupportedCharsDatastore(s string) error {
+	re := regexp.MustCompile(unsuppCharsDatastoreRegex)
+	return checkUnsupportedChars(s, re)
+}
+
+func checkUnsupportedChars(s string, re *regexp.Regexp) error {
+	st := []byte(s)
```

Comment:
I wonder if vSphere supports unicode in path names? if so, since you are looking for special characters in a byte array, you may find many if "not allowed" due to how UTF-8 works.